src/TransformBase.cs

```diff
@@ -0,0 +1,555 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Management.Automation;
+using System.Text;
+
+using Microsoft.PowerShell.PlatyPS.Model;
+
+namespace Microsoft.PowerShell.PlatyPS
+{
+    internal abstract class TransformBase
+    {
+        protected readonly TransformSettings Settings;
+
+        public TransformBase(TransformSettings settings) => Settings = settings;
+
+        internal abstract Collection<CommandHelp> Transform(string[] source);
+
+        protected CommandHelp ConvertCmdletInfo(CommandInfo commandInfo)
+        {
+            string cmdName = commandInfo is ExternalScriptInfo ? commandInfo.Source : commandInfo.Name;
+
+            Collection<PSObject> help = PowerShellAPI.GetHelpForCmdlet(cmdName, Settings.Session);
+
+            bool addDefaultStrings = false;
+            dynamic? helpItem = null;
+
+            if (help.Count == 1)
+            {
+                helpItem = help[0];
+
+                // If the description and examples are empty the help is auto-generated.
+                // So assume that no existing help content is available.
+                if (string.IsNullOrEmpty(GetStringFromDescriptionArray(helpItem.description)) &&
+                    string.IsNullOrEmpty(helpItem.examples))
+                {
+                    addDefaultStrings = true;
+                }
+            }
+            else
+            {
+                addDefaultStrings = true;
+            }
+
+            CommandHelp cmdHelp = new(commandInfo.Name, commandInfo.ModuleName, Settings.Locale);
+            cmdHelp.OnlineVersionUrl = Settings.OnlineVersionUrl;
+            cmdHelp.Synopsis = GetSynopsis(helpItem, addDefaultStrings);
+            cmdHelp.AddSyntaxItemRange(GetSyntaxItem(commandInfo, helpItem));
+            cmdHelp.Description = GetDescription(helpItem, addDefaultStrings);
+            cmdHelp.AddExampleItemRange(GetExamples(helpItem, addDefaultStrings));
+            cmdHelp.AddParameterRange(GetParameters(commandInfo, helpItem, addDefaultStrings));
+
+            cmdHelp.HasCmdletBinding = (commandInfo is FunctionInfo funcInfo && funcInfo.CmdletBinding) ||
+                commandInfo is CmdletInfo ||
+                (commandInfo is ExternalScriptInfo extInfo && extInfo.ScriptBlock.Attributes.Contains(new CmdletBindingAttribute()));
```

Comment:
A `test.ps1` file with the following content is also considered having 'cmdletbinding'. 
```
param(
  [parameter()]
  $Path
)
```
I don't know a good way to get this information for an `ExternalScriptInfo`. I think we should add the `CmdletBinding` property to it too.